contracts/PetPark.sol

```diff
@@ -4,4 +4,112 @@ pragma solidity ^0.8.0;
 
 contract PetPark {
 
+    address owner;
+
+    enum AnimalType { None, Fish, Cat, Dog, Rabbit, Parrot }
+    enum Gender { Male, Female }
+
+    mapping(AnimalType => int) animalCount;
+    mapping(address => AnimalType) isBorrowing;
+
+    event Added(AnimalType animal, int count);
+    event Borrowed(AnimalType animal);
+    event Returned(AnimalType animal);
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    modifier onlyOwner() {
+        require(owner == msg.sender, "Not an owner");
+        _;
+    }
+
+    modifier isOldEnough(uint age) {
+        require(age != 0, "Invalid Age");
+        _;
+    }
+
+    modifier isValidAnimal(AnimalType animal) {
+        require(animal != AnimalType.None, "Invalid animal");
+        _;
+    }
+
+    modifier isValidAnimal2(AnimalType animal) {
+        require(animal != AnimalType.None, "Invalid animal type");
+        _;
+    }
+
+    modifier isAvailableAnimal(AnimalType animal, int count) {
+        require(animalCount[animal] - count >= 0, "Selected animal not available");
+        _;
+    }
+
+    modifier noBoysAllowed(Gender gender, AnimalType animal) {
```

Comment:
You shouldn't create a modifier for the check for "which pets men can borrow" since this condition does not check for men and women both and hence is not a generic condition for borrow function. In your case it fails because condition in this modifier returns false while it should return true for all women